springdoc-openapi-starter-common/src/main/java/org/springdoc/core/providers/HateoasHalProvider.java

```diff
@@ -79,8 +82,26 @@ protected void init() {
 	 */
 	public boolean isHalEnabled() {
 		return hateoasPropertiesOptional
-				.map(HateoasProperties::getUseHalAsDefaultJsonMediaType)
+				.map(this::isHalEnabled)
 				.orElse(true);
 	}
 
+	private boolean isHalEnabled(@NonNull HateoasProperties hateoasProperties) {
+		// In spring-boot 3.5, the method name was changed from getUseHalAsDefaultJsonMediaType to isUseHalAsDefaultJsonMediaType
+		var possibleMethodNames = List.of("isUseHalAsDefaultJsonMediaType", "getUseHalAsDefaultJsonMediaType");
+
+		for (var methodName : possibleMethodNames) {
+			var method = ReflectionUtils.findMethod(HateoasProperties.class, methodName);
+			if (method != null) {
+				var result = ReflectionUtils.invokeMethod(method, hateoasProperties);
+				if (result instanceof Boolean) {
+					return (boolean) result;
```

Comment:
Should be possible to avoid the type-casting here as the project depends on java 17
```suggestion
				if (result instanceof Boolean halEnabled) {
					return halEnabled;
```

Reply:
pushed.